src/annotator/index.js

```diff
@@ -36,32 +36,38 @@ const appLinkEl = /** @type {Element} */ (
 
 function init() {
   const annotatorConfig = getConfig('annotator');
-  const isPDF = typeof window_.PDFViewerApplication !== 'undefined';
+  const sidebarConfig = getConfig('sidebar');
+  const { sidebarAppUrl } = sidebarConfig;
+
+  const frameConnector = new FrameConnector({
+    sidebarAppUrl,
+  });
+  let hostPort = frameConnector.getPort({
+    channel: 'hostToSidebar',
+    port: 'host',
+  });
+
+  if (annotatorConfig.subFrameIdentifier) {
+    // Other modules use this to detect if this frame context belongs to hypothesis.
+    // Needs to be a global property.
+    Object.defineProperty(_window, '__hypothesis_frame', { value: true });
+    hostPort = null;
+  }
+
+  const isPDF = typeof _window.PDFViewerApplication !== 'undefined';
 
-  let frameConnector = /** @type {FrameConnector?} */ (null);
   let sidebar = /** @type {Sidebar?} */ (null);
   let notebook = /** @type {Notebook?} */ (null);
 
   const eventBus = new EventBus();
-  const guest = new Guest(document.body, eventBus, {
+  const guest = new Guest(document.body, eventBus, hostPort, {
     ...annotatorConfig,
     // Load the PDF anchoring/metadata integration.
     // nb. documentType is an internal config property only
     documentType: isPDF ? 'pdf' : 'html',
   });
 
-  if (annotatorConfig.subFrameIdentifier) {
-    // Other modules use this to detect if this
-    // frame context belongs to hypothesis.
-    // Needs to be a global property that's set.
-    window_.__hypothesis_frame = true;
```

Comment:
I would really like to get rid of this global. From what I can see it is used to detect whether the client has already been loaded into a frame. There are other ways we could do that, such as checking for the `<link>` tags that the boot script adds to the page.

Reply:
Sure.